homeassistant/components/lg_soundbar/media_player.py

```diff
@@ -105,7 +105,8 @@ def handle_event(self, response):
                 self._equaliser = data["i_curr_eq"]
             if "s_user_name" in data:
                 self._name = data["s_user_name"]
-        self.schedule_update_ha_state()
+        if self.hass:
```

Comment:
Can you instead attach the handle event function inside `async_added_to_hass` callback?

Reply:
Hi @balloob , 
would you be so kind as to give us an example of this type of implementation? This is a typical event callback 'problem' and will be helpful if we can review a 'good' implementation to follow.
The current implementation for RFLink could benefit from that same solution.

Thanks in advance.

Reply:
Heos does it correctly:
https://github.com/home-assistant/home-assistant/blob/9e79920c9c65bc343ccb3a15fa59588c1eba304c/homeassistant/components/heos/media_player.py#L104-L130

Also make sure to remove the call to `update` in init method. It's already called during entity addition, since `add_entities` is called with second argument `True`.